Libraries/LibWeb/Credentials/CredentialsContainer.cpp

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2024, Miguel Sacristán <miguel_tete17@hotmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "CredentialsContainer.h"
+
+#include <AK/Vector.h>
+#include <LibWeb/Credentials/Credential.h>
+#include <LibWeb/HTML/Window.h>
+#include <LibWeb/Platform/EventLoopPlugin.h>
+
+namespace Web::Credentials {
+GC_DEFINE_ALLOCATOR(CredentialsContainer);
+
+WebIDL::ExceptionOr<GC::Ref<CredentialsContainer>> CredentialsContainer::construct_impl(JS::Realm& realm)
+{
+    return realm.create<CredentialsContainer>(realm);
+}
+
+// https://www.w3.org/TR/credential-management-1/#abstract-opdef-request-a-credential
+GC::Ref<WebIDL::Promise> CredentialsContainer::get(CredentialRequestOptions const& options)
+{
+    // 1. Let settings be the current settings object.
+    auto& settings = HTML::current_principal_settings_object();
+
+    // 2. Assert: settings is a secure context.
+    ASSERT(HTML::is_secure_context(settings));
+
+    // 3. Let document be settings’s relevant global object's associated Document.
+    auto const& document = verify_cast<HTML::Window>(settings.global_object()).associated_document();
+
+    auto p = WebIDL::create_promise(realm());
+
+    // 4. If document is not fully active, then return a promise rejected with an "InvalidStateError" DOMException.
+    if (!document.is_fully_active()) {
+        WebIDL::reject_promise(realm(), p, WebIDL::InvalidStateError::create(realm(), "Document is not fully active yet"_string));
```

Comment:
This can be `return WebIDL::create_rejected_promise_from_exception(realm, WebIDL::FooError::create(...))`

https://github.com/LadybirdBrowser/ladybird/blob/master/Libraries/LibWeb/WebIDL/Promise.h#L35

Reply:
Didn't knew the helper existed, nice